### PR TITLE
Add rustsec audit-check github action support

### PR DESCRIPTION
**Pull request by jstavh**
Add rustsec audit-check github action support

### CODE REVIEW

File: .github/workflows/rustsec.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+name: RustSec Audit Check
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
```
Comment (trevorbernard) [inline .github/workflows/rustsec.yml:5]: I would do a security check any time the cargo.toml or cargo.lock files change.

```yaml
on:
  push:
    paths:
      - '**/Cargo.toml'
      - '**/Cargo.lock'
  ```

File: .github/workflows/rustsec.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+name: RustSec Audit Check
+
```
Comment (trevorbernard) [inline .github/workflows/rustsec.yml:2]: You'll need to set the working directory

```yaml
defaults:
  run:
    working-directory: server
```

**Review by trevorbernard — CHANGES_REQUESTED** [review]

File: .github/workflows/rustsec.yml (hunk 1 above)
Comment (jstavh) [reply .github/workflows/rustsec.yml:5]: good idea

File: .github/workflows/rustsec.yml (hunk 2 above)
Comment (jstavh) [reply .github/workflows/rustsec.yml:2]: good catch :finger_guns:

File: .github/workflows/rustsec.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+on:
+  push:
+    paths:
+      - '**/Cargo.toml'
+      - '**/Cargo.lock'
+  pull_request:
+    types: [opened, synchronize, reopened]
+
```
Comment (trevorbernard) [inline .github/workflows/rustsec.yml:10]: I would remove this altogether or add the paths into the pull request as well

**Review by trevorbernard — CHANGES_REQUESTED** [review]

File: .github/workflows/rustsec.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+      - name: Check out code
+        uses: actions/checkout@v2
+      - name: Setup Rust
+        uses: actions/setup-rust@v2
+        with:
+          profile: minimal
+      - name: Install RustSec
+        run: cargo install rustsec
+      - name: Run RustSec Audit Check
+        run: rustsec audit-check
```
[the remark this answers is not in the document]
Comment (jstavh) [reply .github/workflows/rustsec.yml:26]: Okay will do. Why though?

**Review by trevorbernard — APPROVED** [review]

**Comment by trevorbernard** [comment]
👍🏽

**Pull request merged**